### PR TITLE
Alerting: Send alerts to a remote Alertmanager

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR implements sending alerts to the remote Alertmanager by holding a queue in memory and flushing batches of alerts.

Related issue: https://github.com/grafana/grafana/issues/76692

### CODE REVIEW

File: pkg/services/ngalert/remote/labels.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package remote
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/labels.go:1]: Taken from https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/sender/sender.go

File: pkg/services/ngalert/remote/alertmanager.go [hunk 2]
```diff
@@ -65,30 +73,44 @@ func NewExternalAlertmanager(cfg ExternalAlertmanagerConfig, orgID int64) (*Exte
...
+		log:           logger,
 		url:           cfg.URL,
 		tenantID:      cfg.TenantID,
 		orgID:         orgID,
 		defaultConfig: cfg.DefaultConfig,
+		manager:       manager,
 	}, nil
 }
 
-func (am *ExternalAlertmanager) ApplyConfig(ctx context.Context, config *models.AlertConfiguration) error {
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager.go:79]: A big part of the diff is this name change.
I changed the name of the struct, `remote.ExternalAlertmanager` is redundant. The same goes for file names.

File: pkg/services/ngalert/remote/alertmanager.go [hunk 3]
```diff
@@ -238,30 +267,44 @@ func (am *ExternalAlertmanager) GetReceivers(ctx context.Context) ([]apimodels.R
...
 }
 
-func (am *ExternalAlertmanager) TestTemplate(ctx context.Context, c apimodels.TestTemplatesConfigBodyParams) (*notifier.TestTemplatesResults, error) {
+func (am *Alertmanager) TestTemplate(ctx context.Context, c apimodels.TestTemplatesConfigBodyParams) (*notifier.TestTemplatesResults, error) {
 	return &notifier.TestTemplatesResults{}, nil
 }
 
-func (am *ExternalAlertmanager) StopAndWait() {
+// TODO: change implementation, this is only useful for testing other methods
+func (am *Alertmanager) Ready() bool {
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager.go:279]: Quick readiness check, this will be changed for a more definitive implementation later.

**Comment by gotjosh** [comment]
As dicussed offline, two things I'd like to see in this PR:

1. Let's make sure we extract the rename to a separate PR
2. I'd rather not have us copy the notifier and then use it as is. Let's create an issue to upstream the current patch and perhaps create one where you give it an alertmanager w/o the need to have service discovery.

**Comment by santihernandezc** [comment]
Closed in favor of https://github.com/grafana/grafana/pull/77034

**Pull request closed without merging**